developer/styleguide.md

```diff
@@ -1,119 +1,216 @@
 
-Pull request branches are evaluated using [BitHound](https://www.bithound.io/github/reactioncommerce/reaction) for insecure dependencies and code quality.
+```js
+reaction-paypal
+reaction-google-analytics
+yourorg-your-package
+```
 
-- must pass a Linting check and *no errors* are accepted.
-- must pass a Dependency check and no priority packages are accepted.
-- must pass `reaction test`
+### Group related files together into folders
 
-In many cases, documentation updates can be required as well.
+As a convention, if there are multiple files that provide functionality that is broken down across multiple files, these files should be grouped within a folder. If the files are stand-alone and the name needs to represent functionality, you can use a hyphen for the separator.
 
-Pull requests are submitted to a peer code review process before acceptance.
+**Do**
 
-### File Naming Conventions
+```sh
+/container/settings.js
+/container/settings.html
+/container/ <more files>
+```
+
+or
 
-In general we use hyphens (-) and camelCase for folder names, and camelCase alone for file names. Underscores are not to be used for file or folder names unless expressly required.  Be aware that not all operating systems are case sensitive, so it's not ok to have two files named the same with differing case.
+```sh
+/settings-container.js
+/settings-container.html
+```
 
-#### Folder Names
+**Don't**
 
-**Good**
+```sh
+/container/settings-containers.js
+/container/settingsContainer.html
+settingsContainer.js
+```
 
-- Folder names for packages must contain only lowercased alpha numeric characters and may be hyphenated if joining more than one word
-- Folder names all normal directories must start with a lowercased letter and may camel cased if joining more than one word
+## JavaScript style recommendations
 
+## Variables
 
-    // Packages in /imports/plugins
-    ui-grid/
-    example-paymentmethod/
-    social/
-    taxes-avalara/
+### Naming variables
 
-    // All other folder names everywhere
-    addressBook/
+Variable names should be:
 
-**Bad**
+- Meaningful. Avoid single-letter variables.
+- Use `error`, instead of `e` or `err`
+- Variables that contains a boolean value should use a `isThisTrue` style.
+- Variables that return arrays should be plural.
 
-- Package name should contain hyphens to make it easier to read.
-- Underscores are not to be used unless expressly required.
+Publication names should:
 
+- Use TitleCase
 
-    reactionpackagename/
-    address_book/
+### Working with variables
 
-#### File Names
+- Use `const` except when a variable is being reassigned.
 
-**Good**
+## Methods
 
-- File names must start with a lowercased letter and may be camel cased if joining more than one word.
-- File names may contain multiple (.) characters as needed
+### Method names should:
+- Methods names should use a verb/noun style when possible, e.g. `checkConfiguration` or `addToCart`.
```

Comment:
Not sure if I've written this down anywhere yet, but I think in cases where we might have multiple methods acting on the same resource we should consider the following style: `cart/items/add`, `cart/items/remove` - 
this is probably too much change for this PR. One of the reasons for moving to a style like this is that it will naturally organize our method jsdocs.

I've used this style in the Shopify connector which we also used to experiment with jsdocs.